### PR TITLE
Api fixes

### PR DESCRIPTION
**Pull request by matpiotrowski**
Fixes: #309
Minor fixes

### CODE REVIEW

**Review by jszemplinski — APPROVED** [review]

File: tensorhive/controllers/nodes.py [hunk 1]
```diff
@@ -35,6 +38,33 @@ def get_infrastructure():
...
+                    not_allowed_gpus = all_gpus - allowed_gpus
+                    for key in not_allowed_gpus:
+                        del gpu_list[key]
+                if gpu_list is None or len(gpu_list) == 0:
+                    not_allowed_hostnames.append(hostname)
+            for hostname in not_allowed_hostnames:
+                del infrastructure[hostname]
+        except NoResultFound:
+            # Such user does not exist
+            return {}
```
Comment (roscisz) [inline tensorhive/controllers/nodes.py:66]: Consider extracting this logic to a separate function. All other functions in this controller call get_infrastructure, but maybe a filter_infrastructure_usable_by_user(infrastructure, user_id) could it be useful in other controllers and should be provided by the restriction or user model?

**Comment by roscisz** [comment]
The infrastructure filtering brings problems when switching between users in one browser:
- we have server A and server B, user 1 has access to server A, while user 2 has access to both servers A and B
- log in as user 2, good - I can see both servers in calendar and monitoring
- log in as user 1, good - I can see only server A in calendar and monitoring
- log in again as user 2, error - I can see only server A, and the following error is logged:

```
Exception in thread MonitoringService_Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/tensorhive/TensorHive/tensorhive/core/utils/StoppableThread.py", line 18, in run
    self.do_run()
  File "/home/tensorhive/TensorHive/tensorhive/core/services/MonitoringService.py", line 41, in do_run
    monitor.update(self.connection_manager.connections, self.infrastructure_manager)
  File "/home/tensorhive/TensorHive/tensorhive/core/monitors/GPUMonitor.py", line 17, in update
    self._update_processes(infrastructure_manager, processes)
  File "/home/tensorhive/TensorHive/tensorhive/core/monitors/GPUMonitor.py", line 240, in _update_processes
    if infrastructure_manager.infrastructure[hostname]['GPU'][uuid]['processes'] is None:
KeyError: 'GPU-c56d05e0-4047-135b-325f-6ce728621b71'

On the UI side I can see 404 on the endpoint: http://<secret>/api/nodes/<server>/gpu/metrics
where <server> is server B.
```

File: tensorhive/controllers/nodes.py (hunk 1 above)
Comment (matpiotrowski) [reply tensorhive/controllers/nodes.py]: Fixed in https://github.com/roscisz/TensorHive/pull/311/commits/080b84f872146a2b8c937323073035e670f73508

**Comment by matpiotrowski** [comment]
> The infrastructure filtering brings problems when switching between users in one browser:
> 
>     * we have server A and server B, user 1 has access to server A, while user 2 has access to both servers A and B
> 
>     * log in as user 2, good - I can see both servers in calendar and monitoring
> 
>     * log in as user 1, good - I can see only server A in calendar and monitoring
> 
>     * log in again as user 2, error - I can see only server A

Fixed in https://github.com/roscisz/TensorHive/pull/311/commits/b9fde34cb54e929321c1c1fedcc4b63431d5c98a

**Pull request merged**